### PR TITLE
Integrating Mobile UI and Casting

### PR DESCRIPTION
**Pull request by hghazzi**
Created a new branch that has all the latest changes made by @madikarizma, including his initial updates to support casting.

JW7-3649

### CODE REVIEW

**Comment by robwalch** [comment]
feature/aircast only needs to be in sync with master. anything else you put in  "feature/maui-cast" is unfinished work and may not merge back into master without conflicts.

**Comment by robwalch** [comment]
@hghazzi please resolve the merge conflicts

File: src/css/flags/audioplayer.less [hunk 1]
```diff
@@ -42,6 +42,11 @@
             display: none;
         }
     }
+    &.jw-flag-user-inactive {
```
Comment (robwalch) [inline src/css/flags/audioplayer.less]: @madikarizma @jw-kaurand @hghazzi 

Shouldn't `:not(.jw-flag-audio-player)`be used in `jw-flag-user-inactive`?

File: src/css/flags/cast-available.less [hunk 2]
```diff
@@ -1,10 +1,14 @@
 .jwplayer.jw-flag-cast-available {
+
     .jw-controlbar {
-      display: table;
+        display: table;
```
Comment (robwalch) [inline src/css/flags/cast-available.less]: There are too many places that display is defined for controlbar.

File: src/css/flags/user-inactive.less [hunk 3]
```diff
@@ -9,12 +10,26 @@
     }
 
     &.jw-flag-casting {
-        .jw-controlbar {
-            display: table;
-        }
+      .jw-controlbar {
+        display: table;
```
Comment (robwalch) [inline src/css/flags/user-inactive.less]: again?

**Comment by hghazzi** [comment]
test this please

File: src/css/flags/cast-available.less (hunk 2 above)
Comment (madikarizma) [inline src/css/flags/cast-available.less]: Redundancy removed here: https://github.com/jwplayer/jwplayer/pull/1661/commits/b4ed0e538a6601ebc86d250e3696ab7308e259f3

Found we already accommodated for this in the idle state file.

File: src/css/flags/audioplayer.less (hunk 1 above)
Comment (madikarizma) [inline src/css/flags/audioplayer.less]: Addressed here: https://github.com/jwplayer/jwplayer/pull/1661/commits/ba80e4724c5a38f300e491d19ed7650f24372cb8

File: src/css/flags/user-inactive.less (hunk 3 above)
Comment (madikarizma) [inline src/css/flags/user-inactive.less]: Addressed: https://github.com/jwplayer/jwplayer/pull/1661/commits/ba80e4724c5a38f300e491d19ed7650f24372cb8

**Comment by madikarizma** [comment]
@robwalch Addressed your comments in https://github.com/jwplayer/jwplayer/pull/1661/commits/ba80e4724c5a38f300e491d19ed7650f24372cb8 and https://github.com/jwplayer/jwplayer/pull/1661/commits/b4ed0e538a6601ebc86d250e3696ab7308e259f3

File: src/js/view/view.js [hunk 4]
```diff
@@ -401,6 +402,7 @@ define([
 
             _videoLayer = _playerElement.getElementsByClassName('jw-media')[0];
 
+            _mediaLayer = _playerElement.querySelector('.jw-media');
```
Comment (robwalch) [inline src/js/view/view.js]: What is the difference between `_videoLayer` and `_mediaLayer`? It's the same element:

`getElementsByClassName('jw-media')[0] === querySelector('.jw-media')`

File: src/js/view/view.js [hunk 5]
```diff
@@ -1117,6 +1119,10 @@ define([
             return _displayClickHandler;
         };
 
+        this.mediaContainer = function() {
+            return _mediaLayer;
+        };
```
Comment (robwalch) [inline src/js/view/view.js]: Remove. This is already available in the model.

File: src/css/imports/cast.less [hunk 6]
```diff
@@ -1,69 +1,48 @@
...
+@keyframes jw-cast-text {
+  0% {
+    opacity: 1;
+  }
+  50% {
+    opacity: 0.5;
+  }
+  100% {
+    opacity: 1;
+  }
```
Comment (robwalch) [inline src/css/imports/cast.less]: Please remove the pulsing text.

**Pull request merged**